### PR TITLE
fix return value and empty requests

### PR DESCRIPTION
**Pull request by Kriechi**

### CODE REVIEW

File: libmproxy/protocol2/http.py [hunk 1]
```diff
@@ -237,6 +238,9 @@ def __call__(self):
 
                 self.log("request", "debug", [repr(request)])
 
+                if not request:
+                    return
+
```
Comment (mhils) [inline libmproxy/protocol2/http.py]: Why is this part neccessary?

Comment (Kriechi) [inline libmproxy/protocol2/http.py]: just a safe guard...

Comment (mhils) [inline libmproxy/protocol2/http.py]: Ok. Let's keep that out -`HTTPRequest.from_protocol` should raise a NetLibError if it is unable to return a request, which we catch above.

Comment (Kriechi) [inline libmproxy/protocol2/http.py]: fixed.

**Pull request merged**